server/consumer.go

```diff
@@ -1680,12 +1680,14 @@ func (o *consumer) processNextMsgReq(_ *subscription, c *client, _, reply string
 	if wr.noWait {
 		if o.maxp > 0 && len(o.pending) >= o.maxp {
 			sendErr(409, "Exceeded MaxAckPending")
+			return
```

Comment:
Unlock first?

Reply:
Also, same Unlock missing here: https://github.com/nats-io/nats-server/blob/b9498b8c0e0983e9c72eb8f0a3494fb76b1fff02/server/consumer.go#L1673

Reply:
sendErr does it.

Reply:
Oh, sendErr is unlocking, right?